main.tf

```diff
@@ -111,6 +111,20 @@ resource "kubernetes_stateful_set_v1" "relay" {
           }
           image_pull_policy = "Always"
         }
+
+        affinity {
```

Comment:
@haroldsphinx As we discussed earlier in the meeting, let's make the affinity rule a variable (for two reasons: backward compatibility and to be cloud provider agnostic).
This can be done like this:

`variable "node_affinity_config" {
  description = "Node Affinity configuration"
  type        = map(any)
  default     = {}
}
`

`dynamic "affinity" {
      for_each = var.node_affinity_config != {} ? [1] : []
      content {
        node_affinity = var.node_affinity_config
      }
    }`
   
  Then we pass the zone-specific affinity we need from the relay-2 downstream.
  
  Also, we have to run the relay in a single Az and tune the affinity rule to achieve this. After that, remove the existing relay-2 statefulset and PVs, deploy the new relay-0 with the affinity rules, and make sure both pods and PVs are created in a single zone.